src/otx/core/model/entity/base.py

```diff
@@ -251,6 +266,10 @@ def _export_parameters(self) -> dict[str, Any]:
 
         return parameters
 
+    def _get_forward_with_auxiliaries(self) -> Callable | None:
```

Comment:
Why don't we just add `forward_explain()`  in this PR? It seems getting dirty without this I think.

Reply:
If we go this way, we also have to provide a native mechanism for setting explain mode for the model. Now, it's done with a hook, but this mechanism has nothing to do with export and `forward_explain()`. So I believe it should be like explain_mode set -> forward is replaced with `forward_explain()`. At the same time, this should not be conflicting with the existing loss/predict modes.

Reply:
Now it looks like custom forward immediately breaks mm's predict/loss method, while hooks allows to leave it as is.
Having this in mind, Evgeny's solution looks appropriate (replacing forward with xai one only for export), unless we change forward -> predict/loss -> _customize_outputs pipeline to support XAI natively

Reply:
No, I just wanna add `forward_explain()`, not overriding `forward()`. Then, use this function for 1) `OTXModel.export(with_xai=True) and 2) `Engine.predict(with_xai=True)`.